mlflow/spark.py

```diff
@@ -103,9 +107,36 @@ def log_model(spark_model, artifact_path, conda_env=None, jars=None, dfs_tmpdir=
     >>> model = pipeline.fit(training)
     >>> mlflow.spark.log_model(model, "spark-model")
     """
-    return Model.log(artifact_path=artifact_path, flavor=mlflow.spark, spark_model=spark_model,
-                     jars=jars, conda_env=conda_env, dfs_tmpdir=dfs_tmpdir,
-                     sample_input=sample_input)
+    _validate_model(spark_model, jars)
+    run_id = mlflow.tracking.fluent._get_or_start_run().info.run_uuid
+    run_root_artifact_uri = mlflow.get_artifact_uri()
+    # If the artifact URI is a local filesystem path, defer to Model.log() to persist the model
+    if mlflow.tracking.utils._is_local_uri(run_root_artifact_uri):
```

Comment:
Could we use the optimized logging process for local artifact URIs as well? This would save users an extra filesystem copy and simplify the logical flow of this method.

Reply:
Discussed offline, will add a comment explaining why we use Model.log() for local artifact URIs